torch/utils/data/sampler.py

```diff
@@ -214,7 +226,8 @@ def __iter__(self):
             yield batch
 
     def __len__(self):
+        # Can be called only if the sampler has __len__ implemented
         if self.drop_last:
-            return len(self.sampler) // self.batch_size
+            return len(self.sampler) // self.batch_size  # type: ignore
```

Comment:
What's up with the `# type: ignore` ? Might be good to include the reason as a code comment as well

Reply:
I have just expanded on the comment two lines above.